src/sagemaker/estimator.py

```diff
@@ -670,8 +670,9 @@ def __init__(self, entry_point, source_dir=None, hyperparameters=None, enable_cl
                 training jobs. This will be ignored for now and removed in a further release.
             container_log_level (int): Log level to use within the container (default: logging.INFO).
                 Valid values are defined in the Python logging module.
-            code_location (str): Name of the S3 bucket where custom code is uploaded (default: None).
+            code_location (str): The S3 URI where custom code is uploaded (default: None).
```

Comment:
let's fix this to make it clear we expect a prefix and will append /source/sourcedir.tar.gz